helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -1317,6 +1318,13 @@ public void handleNewSession(String sessionId) throws Exception {
     LOG.info("Handle new session, instance: {}, type: {}, session id: {}.", _instanceName,
         _instanceType,  sessionId);
 
+    // Will only create live instance
+    if (LiveInstance.LiveInstanceStatus.PAUSED
```

Comment:
The change looks good to me. 
Nit: I think usually we follow this pattern when comparing: 
`variable.equal(some_static_value)`

Reply:
Thanks, @xyuanlu, for the review.
This is another coding way to avoid potential NPE. If the variable is null, `variable.equals()` could cause NPE. Then we have to check `if (variable != null && variable.equals())`. Here, using `PAUSED.equals()` helps simplify the code.